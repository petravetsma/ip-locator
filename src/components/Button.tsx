import styled from 'styled-components';

const StyledButton = styled.button`
    vertical-align: top;
    height: 35px;
    margin: 0;
    padding: 15px;
    border: none;
    border-radius: 0 6px 6px 0;
    outline: none;
    background: 50% no-repeat url('./src/assets/right-arrow.svg') #101820;

    &:focus {
        outline: none;
    }
`

type ButtonProps = {
    /**
     * Button click handler.
     */
    onClick: () => void;
}

/**
 * Button with right arrow icon.
 */
export function ArrowButton(props: ButtonProps) {

    return (
        <StyledButton onClick={props.onClick} />
    )
}