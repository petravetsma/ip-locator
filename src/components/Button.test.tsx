import renderer from 'react-test-renderer'
import { Button } from './Button'
import { vi } from 'vitest';

describe('Test button component', () => {

    it('Button rendered with SVG', () => {

        const component = renderer.create(
            <Button onClick={() => null} />,
        )
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('When button clicked, handler invoked', () => {

        const clickHandler = vi.fn();

        const button = renderer.create(
            <Button onClick={clickHandler} />,
        )
        button.root.findByType('button').props.onClick();

        expect(clickHandler).toHaveBeenCalledTimes(1);
    })
})