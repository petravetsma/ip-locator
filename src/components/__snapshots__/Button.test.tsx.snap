// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test button component > Button rendered with SVG 1`] = `
<button
  className="sc-aXZVg bbjbIS"
  onClick={[Function]}
>
  <svg
    className="sc-gEvEer bytOwd"
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      fill="white"
      points="7.293 4.707 14.586 12 7.293 19.293 8.707 20.707 17.414 12 8.707 3.293 7.293 4.707"
    />
  </svg>
</button>
`;
